src/algorithms/path_contexts.py

```diff
@@ -0,0 +1,331 @@
+from collections import OrderedDict
+from bblfsh import Node
+from algorithms.structures.extended_node import ExtNode
+from pprint import pprint
+
+import argparse
+import os
+import fnmatch
+
+
+def recursive_glob(rootdir='.', pattern='*'):
```

Comment:
@kafkasl did you have a look at pathlib (https://docs.python.org/3.4/library/pathlib.html) and it's respective `rlgob()` function?

Reply:
@abs51295 didn't know about it, I'll have a look. However, the glob is just to test it without spark because sourced.ml already handles the glob etc...

Reply:
Okay cool. But also in that case, I think it's better to use the in-built functionality, rather than creating our own. WDYT?

Reply:
yeah sure feel free to change it

Reply:
hety @abs51295 I checked out the lib and it's an order of magnitude slower in some simple tests (only listing files not manipulating path). It's totally worth to do more complex path ops or multi OS though. Thanks for pointing it out.

Reply:
Point noted @kafkasl . Thanks for exploring.